src/mono/mono/mini/mini-posix.c

```diff
@@ -517,48 +518,11 @@ static volatile gint32 sampling_thread_running;
 
 #ifdef HOST_DARWIN
 
-static clock_serv_t sampling_clock_service;
+static clock_serv_t sampling_clock;
 
-static void
-clock_init (MonoProfilerSampleMode mode)
+void
+clock_init_for_profiler (MonoProfilerSampleMode mode)
 {
-	kern_return_t ret;
-
-	do {
-		ret = host_get_clock_service (mach_host_self (), SYSTEM_CLOCK, &sampling_clock_service);
```

Comment:
This is not called anymore.  Is this OK?  Isn't this call under macOS necessary to get the clock you need?

Reply:
It is now part of `mono_clock_init` in mono-time.c.